desc/integrals/quad_utils.py

```diff
@@ -0,0 +1,246 @@
+"""Utilities for quadratures."""
+
+from orthax.legendre import legder, legval
+
+from desc.backend import eigh_tridiagonal, jnp, put
+from desc.utils import errorif
+
+
+def bijection_to_disc(x, a, b):
+    """[a, b] ∋ x ↦ y ∈ [−1, 1]."""
+    y = 2.0 * (x - a) / (b - a) - 1.0
+    return y
+
+
+def bijection_from_disc(x, a, b):
+    """[−1, 1] ∋ x ↦ y ∈ [a, b]."""
+    y = 0.5 * (b - a) * (x + 1.0) + a
+    return y
+
+
+def grad_bijection_from_disc(a, b):
+    """Gradient of affine bijection from disc."""
+    dy_dx = 0.5 * (b - a)
+    return dy_dx
+
+
+def automorphism_arcsin(x):
+    """[-1, 1] ∋ x ↦ y ∈ [−1, 1].
+
+    The arcsin transformation introduces a singularity that augments the singularity
+    in the bounce integral, so the quadrature scheme used to evaluate the integral must
+    work well on functions with large derivative near the boundary.
+
+    Parameters
+    ----------
+    x : jnp.ndarray
+        Points to transform.
+
+    Returns
+    -------
+    y : jnp.ndarray
+        Transformed points.
+
+    """
+    y = 2.0 * jnp.arcsin(x) / jnp.pi
+    return y
+
+
+def grad_automorphism_arcsin(x):
+    """Gradient of arcsin automorphism."""
+    dy_dx = 2.0 / (jnp.sqrt(1.0 - x**2) * jnp.pi)
+    return dy_dx
+
+
+grad_automorphism_arcsin.__doc__ += "\n" + automorphism_arcsin.__doc__
```

Comment:
What does this do?

Reply:
it appends the docstring of the `automorhism_arcsin` function to the docstring of the `grad_automorphism_arcin` function